Tree/README.markdown

```diff
@@ -151,6 +153,149 @@ tree.search("chai")     // returns the "chai" node
 tree.search("bubbly")   // nil
 ```
 
+## Solution 2 (using an enum)
+
+The first solution uses a class to represent the Tree datastructure. Classes are passed by reference in Swift, meaning that if you copy your object and modify it, that will also affect both of your references, because they are referencing the exact same place in data:
+
+```swift
+let sameTree = tree
+sameTree.value = "drinks"
+
+sameTree.value   // "drinks"
+tree.value       // also "drinks"
+```
+
+While this can be desired in some cases, in other cases you want to have something, that is passed by value, meaning that it is copied every time you hand it to a function, assign it to a variable, etc. To implement this, we will use and enum:
+
+```swift
+enum TreeNode<T> {
+  case Leaf(value: T)
```

Comment:
UpperCamelCase for enum cases has been deprecated for lowerCamelCase since Swift 3.

```
enum TreeNode<T> {
  // preferred
  case leaf(value: T)
  indirect case node(value: T, children: [TreeNode])
}
```